readme.md

```diff
@@ -164,52 +170,53 @@ _Optional, Default: `super-secret-key`_
 
 The key used to encrypt session cookie information. You `must` use a strong value here.
 
-##### `startTLS`
+##### `serverList.startTLS`
 _Optional, Default: `false`_
 
 If set to true, instruct `ldapAuth` to issue a `StartTLS` request when initializing the connection with the LDAP server.
 
-##### `insecureSkipVerify`
+##### `serverList.insecureSkipVerify`
 _Optional, Default: `false`_
 
 When connecting to a `ldaps` server or `startTLS` is enabled, the connection to the LDAP server is verified to be secure. This option allows `ldapAuth` to proceed and operate even for server connections otherwise considered insecure.
 
-##### `minVersionTLS`
+##### `serverList.minVersionTLS`
 _Optional, Default: `tls.VersionTLS12`_
```

Comment:
Is this still true ? As per [ldapauth.go L617](https://github.com/wiltonsr/ldapAuth/pull/65/files#diff-d193dd8c19f73d8ebae4e41f0e508fab2a2c7aa6643e32f5ef78912028902035R617), it looks like it will default to VersionTLS10. We probably want to keep the minimum to at least 12.

Reply:
fixed

Reply:
I don't know if we should fix the default value for MinTLSVersion that way as it will now enforce TLS 1.2 as the MaxVersion as well, unless set explicitly. We should keep the MaxTLSVersion to 1.3 by default. (which is also what the docs say.

Maybe we should set it if not defined when the configs are parsed.

Reply:
Setting default values for server-related parameters became a problem with `ServerList`. We can use `CreateConfig` and provide the default values like before, but it will work only for one server once we can't determine how many servers will be provided.

Something like

```golang
[...]
ServerList: []LdapServerConfig{
	LdapServerConfig{
		Url:           "teset.com",
		MinVersionTLS: "tls.VersionTLS12",
		MaxVersionTLS: "tls.VersionTLS13",
	},
},
[...]
```

But I don't like this approach. Do you have any suggestions for reimplementing the default value assignment?

We also have another issue, this one just cosmetic, where the server list parameters are not printed nicely in debug mode.

Reply:
I created a new method `settingDefaults` and refactored `logConfigParams`.

This allows us to set default values to ServerList when necessary, and the DEBUG log is prettified again.